crates/services/consensus_module/poa/src/service.rs

```diff
@@ -131,14 +145,17 @@ where
     T: TransactionPool,
 {
     pub fn new(
-        last_height: BlockHeight,
+        last_block: &BlockHeader,
         config: Config,
         txpool: T,
         block_producer: B,
         block_importer: I,
     ) -> Self {
         let tx_status_update_stream = txpool.transaction_status_events();
         let (request_sender, request_receiver) = mpsc::channel(100);
+        let last_timestamp = last_block.time();
+        let duration = Duration::from_secs(Tai64::now().0 - last_timestamp.0);
```

Comment:
What happens if the service is initialized after some manual blocks were already produced with a timestamp > now?

Reply:
Overflow=) Replaced with `saturating_sub`